### PR TITLE
docs: Avatar component API

### PR DESCRIPTION
**Pull request by snitin315**
### [Formatted Document](https://github.com/razorpay/blade/blob/docs/avatar-api/packages/blade/src/components/Avatar/_decisions/decisions.md)

### CODE REVIEW

File: packages/blade/src/components/Avatar/_decisions/decisions.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+   * The maximum number of avatars to display before truncating.
+   */
+  maxAvatars?: number;
+};
+
+type AvatarProps = {
+  /**
+   * The content or children of the avatar.
+   */
+  children?: StringChildrenType;
```
Comment (Rozamo) [inline packages/blade/src/components/Avatar/_decisions/decisions.md:41]: How is `children` different from `name`?

File: packages/blade/src/components/Avatar/_decisions/decisions.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  icon?: IconComponent;
+  /**
+   * The name of the avatar, used to generate initials.
+   * If src has loaded, the name will be used as the alt attribute of the img. If src is not loaded, the name will be used to create the initials.
+   */
+  name?: string;
+  /**
+   * Custom image source for an image avatar.
+   */
+  src?: string;
```
Comment (Rozamo) [inline packages/blade/src/components/Avatar/_decisions/decisions.md:66]: What is the order of precedence among these three if the consumer provides all of them?

File: packages/blade/src/components/Avatar/_decisions/decisions.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+#### Avatar Fallbacks
+
+If there is an error loading the src of the avatar, there will be 2 fallbacks:
+
+- If there's a `name` prop, we use it to generate the initials.
+  - Avatar will always have 2 letters as initials.
+  - `Rama Krushna Behra` - `RK`
+  - `Nitin` - `NI`
+  - `Anurag Hazra` - `AH`
+- If there's no `name` prop, we use a default avatar.
```
Comment (Rozamo) [inline packages/blade/src/components/Avatar/_decisions/decisions.md:124]: Should we make the default avatar configurable?

File: packages/blade/src/components/Avatar/_decisions/decisions.md (hunk 2 above)
Comment (snitin315) [reply packages/blade/src/components/Avatar/_decisions/decisions.md:66]: `src > icon > name` will be the priority.

File: packages/blade/src/components/Avatar/_decisions/decisions.md (hunk 3 above)
Comment (snitin315) [reply packages/blade/src/components/Avatar/_decisions/decisions.md:124]: Only the `color` is configurable, the  icon remains the same.

Screenshot 2024-05-09 at 2 31 55 PM

**Review by anuraghazra — APPROVED** [review]

**Pull request merged**